src/screens/Exchange/Exchange.js

```diff
@@ -436,9 +437,12 @@ class ExchangeScreen extends React.Component<Props, State> {
   }
 
   componentDidMount() {
-    const { exchangeSearchRequest = {} } = this.props;
+    const { exchangeSearchRequest = {}, baseFiatCurrency } = this.props;
     this.provideOptions();
-    const { fromAssetCode = ETH, toAssetCode, fromAmount } = exchangeSearchRequest;
+    const fiatCurrency = baseFiatCurrency || defaultFiatCurrency;
+
+    const defaultFrom = this.checkIfAssetsExchangeIsAllowed() ? ETH : fiatCurrency;
```

Comment:
Not sure about `checkIfAssetsExchangeIsAllowed` checks it checks for smart wallet feature and if it's enabled then default `from` is ETH? Or you want to check fiat exchange flag here? Then check method renaming needed and current ternary would be reverse?